stl/inc/format

```diff
@@ -3346,8 +3346,16 @@ template <_Format_supported_charT _CharT, class _Traits>
 struct formatter<basic_string_view<_CharT, _Traits>, _CharT>
     : _Formatter_base<basic_string_view<_CharT, _Traits>, _CharT, _Basic_format_arg_type::_String_type> {};
 
+// TRANSITION, VSO-1444086 (remove _STL_DECLSPEC_NOINITALL)
+#if defined(__clang__) || defined(__CUDACC__) || defined(__INTEL_COMPILER) || defined(__EDG__)
+#define _STL_DECLSPEC_NOINITALL
+#else // ^^^ not MSVC/ MSVC vvv
+#define _STL_DECLSPEC_NOINITALL \
+    __pragma(warning(push)) __pragma(warning(disable : 4845)) __declspec(no_init_all) __pragma(warning(pop))

```

Comment:
Since we're excluding Cuda and Intel, we can use Standard `_Pragma`:

https://github.com/microsoft/STL/blob/60decd0d829e68666b556780002c83605d748d80/stl/inc/yvals_core.h#L590-L602